client/bindings/python/iota_client/client.py

```diff
@@ -4,13 +4,99 @@
 from iota_client._high_level_api import HighLevelAPI
 from iota_client._utils import Utils
 from json import dumps
+import humps
+from datetime import timedelta
 
 class IotaClient(NodeCoreAPI, NodeIndexerAPI, HighLevelAPI, Utils):
-    def __init__(self, client_config=None):
+    def __init__(
+        self,
+        nodes=None,
+        primary_node = None,
+        primary_pow_node = None,
+        permanode = None,
+        ignore_node_health = None,
+        api_timeout = None, 
+        node_sync_interval = None,
+        remote_pow_timeout = None,
+        tips_interval = None,
+        quorum = None,
+        min_quorum_size = None,
+        quorum_threshold = None,
+        user_agent = None,
+        local_pow = None,
+        fallback_to_local_pow = None,
+        pow_worker_count = None
+    ):
         """Initialize the IOTA Client.
+
+        Parameters
+        ----------
+        nodes : string or array of string
+            A single Node URL or an array of URLs.
+        primary_node : string
+            Node which will be tried first for all requests.
+        primary_pow_node : string
+            Node which will be tried first when using remote PoW, even before the primary_node.
+        permanode : string
+            Permanode URL.
+        ignore_node_health : bool
+            If the node health should be ignored.
+        api_timeout : datetime.timedelta
+            Timeout for API requests.
+        node_sync_interval : datetime.timedelta
+            Interval in which nodes will be checked for their sync status and the [NetworkInfo](crate::NetworkInfo) gets updated.
+        remote_pow_timeout : datetime.timedelta
+            Timeout when sending a block that requires remote proof of work.
+        tips_interval : int
+            Tips request interval during PoW in seconds.
+        quorum : bool
+            If node quorum is enabled. Will compare the responses from multiple nodes 
+            and only returns the response if `quorum_threshold`% of the nodes return the same one.
+        min_quorum_size : int
+            Minimum amount of nodes required for request when quorum is enabled.
+        quorum_threshold : int
+            % of nodes that have to return the same response so it gets accepted.
+        user_agent : string
+            The User-Agent header for requests.
+        local_pow : bool
+            Local proof of work.
+        fallback_to_local_pow : bool
+            Fallback to local proof of work if the node doesn't support remote PoW.
+        pow_worker_count : int
+            The amount of threads to be used for proof of work.
         """
-        if client_config:
-            client_config = dumps(client_config)
+        client_config = dict(locals())
+        del client_config['self']
+
+        if isinstance(nodes, list):
+            _nodes = nodes
```

Comment:
Can't you just iter over `nodes` directly?

Reply:
Wdym? nodes can be a string or a list of strings 🤔

Reply:
Why do you do `_nodes = nodes` instead of something like this?
```python
for i, node in enumerate(nodes):
    if isinstance(node, Node):
            nodes[i] = node.as_dict()
        else:
            nodes[i] = node
```

Reply:
I'm not a python dev 😆
What you suggest should work. Will change it

Reply:
Hmm, wait. This won't work.

I you call the client like: `client = IotaClient('node-url')` you pass an array of char.
That's why I checked if it is a list first 🤔

Reply:
Tbh currently I don't have any idea why I did it 😅 .
Will check again later

Reply:
Done